### PR TITLE
Add extra functionality and fix classifiers

### PR DESCRIPTION
**Pull request by paddyroddy**
I've basically added stuff to the TOML that I've learnt since last time.

The linting is now being ran at the top level as well as creating a `cookieninja` project and linting in there. I'm going to raise an issue at https://github.com/pappasam/toml-sort about whether we can silence `jinja2` templating.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ repos:
     rev: v0.23.0
     hooks:
       - id: toml-sort-fix
+        exclude: "{{cookiecutter.project_slug}}/pyproject.toml"
```
Comment (paddyroddy) [inline .pre-commit-config.yaml:15]: It's unfortunate, but the invalid syntax throws it off

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -27,7 +28,6 @@ repos:
       - id: check-case-conflict
       - id: check-docstring-first
       - id: check-merge-conflict
-      - id: check-toml
```
Comment (paddyroddy) [inline .pre-commit-config.yaml:30]: Also complains so easier to turn it off at the top level

File: {{cookiecutter.project_slug}}/pyproject.toml [hunk 3]
```diff
@@ -9,26 +9,41 @@ authors = [
...
     "pre-commit",
+    "ruff",
     "tox",
-]
+    "twine",
+]}
+readme = "README.md"
+requires-python = ">={{ cookiecutter.min_python_version }}, <={{ cookiecutter.max_python_version }}"
+version = "{{ cookiecutter.version }}"
+license.file = "LICENCE.md"
```
Comment (paddyroddy) [inline {{cookiecutter.project_slug}}/pyproject.toml:48]: Works with @samcunliffe's changes

File: {{cookiecutter.project_slug}}/pyproject.toml [hunk 4]
```diff
@@ -9,26 +9,41 @@ authors = [
...
+    "Operating System :: POSIX",
+    "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3 :: Only",
+    {%- for python_version in range(
+       cookiecutter.min_python_version | replace('3.', '') | int,
+       cookiecutter.max_python_version | replace('3.', '') | int + 1
+    ) %}
+    "Programming Language :: Python :: 3.{{ python_version }}",
+    {%- endfor %}
     "Typing :: Typed",
```
Comment (paddyroddy) [inline {{cookiecutter.project_slug}}/pyproject.toml:22]: Templating is fine but TOML panics

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -22,9 +22,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          python -m pip install tox tox-gh-actions
+        run: python -m pip install tox tox-gh-actions
```
Comment (samcunliffe) [inline .github/workflows/test.yml:25]: Why don't we need to upgrade pip here?

**Review by samcunliffe — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (paddyroddy) [reply .pre-commit-config.yaml:15]: Issue raised at pappasam/toml-sort#57

File: .github/workflows/test.yml (hunk 5 above)
Comment (paddyroddy) [reply .github/workflows/test.yml:25]: See discussion here https://github.com/actions/setup-python/issues/506#issuecomment-1262522726

**Pull request merged**